### ISSUE TITLE
Game Crashing after (v.2.4.0-v2.4.2) update with DXVK

### ISSUE THREAD

**Issue by SA64BD** [ThirteenAG/GTAIV.EFLC.FusionFix#385]
### Game version this happens in

v1.2.0.59

### What area of the game

![Capture error](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/assets/75331759/05f4f2db-9b2e-4950-a851-5db29e58bf9b)

This crash occurs when attempting to focus the game window after minimizing the game with alt tab. Also Changing resolution results the same problem.
[savedata.zip](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13817303/savedata.zip)

### What should happen

It shouldn't crash.

### CrashDumps and logs

[GTA IV.exe.20240103145752.dmp](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13817158/GTA.IV.exe.20240103145752.dmp)
[GTA IV.exe.20240103145752.log](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13817160/GTA.IV.exe.20240103145752.log)

### Last working version

v2.3.1

### Checklist

- [X] Test in the [latest git build](https://ci.appveyor.com/api/projects/ThirteenAG/gtaiv-eflc-fusionfix/artifacts/GTAIV.EFLC.FusionFix.zip) in case it's already fixed.
- [X] [Search for other reports](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/search?q=my+issue&type=issues) of the same issue.
- [X] Include logs or screenshots of issue.

**Comment by ThirteenAG** [comment]
Can you upload what settings you're using(.cfg and .ini)?

**Comment by SA64BD** [comment]
> Can you upload what settings you're using(.cfg and .ini)?
 
Sure, here you go
[cfg and ini.zip](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13817375/cfg.and.ini.zip)

**Comment by ThirteenAG** [comment]
Unfortunately can't reproduce, can you give more context, it happens under certain circumstances maybe?

**Comment by ThirteenAG** [comment]
Also you have 
![image](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/assets/4904157/a078f678-8dcf-4f55-8c0f-e118b49aa281)
and some other mods, possible that it's not Fusion Fix related.

**Comment by SA64BD** [comment]
I removed 'dxwrapper.asi' and the problem occurs with 'DXVK GPL Async (2.3-1)'. Without DXVK, the game works fine. But earlier versions of fusionfix (v2.3.1) did not had such issue.

[DXVK GPL Async 2.3-1.zip](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13817680/DXVK.GPL.Async.2.3-1.zip)

**Comment by SA64BD** [comment]
is it possible that newer versions of fusionfix may have a conflict with dxvk?

**Comment by RecklessGlue540** [comment]
I also have this issue ever since I've upgraded my GPU.
It's a problem with the drivers maybe... Using Radeon?
Also do not use DXVK with dxwrapper. One or the other.
I've seen DXWrapper should fix it. DXVK doesn't fully tho.

**Comment by SA64BD** [comment]
> I also have this issue ever since I've upgraded my GPU. It happens when changing graphics settings It's a problem with the drivers maybe... Using Radeon? Also do not use DXVK with dxwrapper. One or the other. I've seen DXWrapper should fix it. DXVK doesn't fully tho.

I have removed both dxwrapper and DXVK and then tried adding back only DXVK. Still causes the issue.

**Comment by RecklessGlue540** [comment]
Try DXWrapper only

**Comment by SA64BD** [comment]
> Try DXWrapper only

DXwrapper works fine, still It would be very greatful for being able to play with DXVK again, just like the previous version of Fusionfix (v2.3.1). Since it really helps with the performance and now has became essential way to play the game atm

**Comment by ThirteenAG** [comment]
We're looking into it with @RaphaelK12

**Comment by RecklessGlue540** [comment]
> We're looking into it with @RaphaelK12

Really? Amazing. Magic AI reported this as well in the past..

**Comment by RecklessGlue540** [comment]
@ThirteenAG, I should also mention this also happens vanilla. Not a FusionFix bug. For me at least.
Only thing that apparently makes it go away is DXWrapper and the thing randomly happens when changing ANY graphics setting.
Not the ones introduced with FusionFix tho. Those don't "Reset the Direct3D Device"...
DXVK further randoms the error.

**Comment by ThirteenAG** [comment]
[GTAIV.EFLC.FusionFix.zip](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13821395/GTAIV.EFLC.FusionFix.zip)

I can't reproduce, so try this build I guess.

**Comment by RecklessGlue540** [comment]
> [GTAIV.EFLC.FusionFix.zip](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13821395/GTAIV.EFLC.FusionFix.zip)
> 
> I can't reproduce, so try this build I guess.

Didn't work on just D3D9.

**Comment by ThirteenAG** [comment]
I'm debugging a regression from v2.3.1, what d3d9 has to do with it.

**Comment by RecklessGlue540** [comment]
@SA64BD, can you try and see if you get the issue without any wrapper? Just vanilla? I feel like I have a totally different issue even tho the error is the exact same.

**Comment by Mondea** [comment]
Been having the same issue drive me crazy. Was having the problem when using DXVK v2.3 and the latest FusionFix. Would get D3D error when changing any graphics settings.
Then did a complete reinstall of the game on steam, and tested only with DXVK v2.2. Ran very good, no crashes when changing settings. Very good performance (90FPS at 5120x2880). 
After verifying that DXVK works, I installed the latest FusionFix. This immediately dropped the frames to barely 60FPS at 4K.  The game won't crash when changing graphics settings anymore, but it does after a while of playing. 

Using Lenovo Legion 5 Pro 3070 12700H with the game on dGPU (3070) monitor.

**Comment by SA64BD** [comment]
I just tested the game from vanilla. According to my finding, It seems like the latest fusionfix works with dxvk. However whenever i add project2dfx mod, the game won't crash on first run. But then closing the game and running the game on 2nd time, causes this issue. I also faced where the game crashed on 1st attempt while didn't crashed on 2nd attempt with same project2dfx installed. I also tried the testing with other mods like ConsoleSelectMenuIV.asi, HitmarkerIV.asi, IVLodLights, RadioWheelIV.asi, RestoredDeathMusivIV.asi, SimpleDotIV.asi.

In conclusion
Vanilla+FusionFixLatest=✔️
Vanilla+FusionFixLatest+DXVK=✔️(unsure since @Mondea mentioned his issue previously)
Vanilla+FusionFixLatest+Mods=✔️
Vanilla+FusionFixLatest+DXVK+Mods=❌
Vanilla+FusionFix(v2.3.1)+DXVK+Mods=✔️

FusionFix didn't broke the mods, neither it broke DXVK indiviually. But combining mods with DXVK breaks the game.

**Comment by RecklessGlue540** [comment]
Mine crashes all of the time on vanilla D3D9 when changing graphics settings. FusionFix doesn't influence this in any way.
DXVK is rather random. I got it a few times, now I can't reproduce anymore...
DXWrapper fixes this completely from what I've seen.

I think I got some useful info from a very helpful guy on the Fusion Team saying that the game uses DDraw as fallback to query the VRAM and GPU, in case DXGI and NVAPI won't or can't. Being on AMD Radeon myself, NVAPI isn't a thing here afaik and I don't think IV uses DXGI either...
Maybe DDraw doesn't do it's job properly resetting the Direct3D device when changing settings thus crashing? @ThirteenAG, what do you think? Tho I'm pretty (actually very) sure I'm saying absurd bullshit, it makes some sense for me, I dunno...

**Comment by RaphaelK12** [comment]
[GTAIV.EFLC.FusionFix_test_01.zip](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13824568/GTAIV.EFLC.FusionFix_test_01.zip)

@RecklessGlue540  Try if this works, I believe I fixed this problem and another possible one, both related to the device reset failure.

**Comment by RecklessGlue540** [comment]
> [GTAIV.EFLC.FusionFix_test_01.zip](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13824568/GTAIV.EFLC.FusionFix_test_01.zip)
> 
> @RecklessGlue540 Try if this works, I believe I fixed this problem and another possible one, both related to the device reset failure.

Didn't seem to work...

**Comment by RecklessGlue540** [comment]
Might just be a driver issue?

**Comment by RaphaelK12** [comment]
Yes, first attempt I got an error with RivaTuner.
I spent more than an hour changing settings, switching between windowed and full screen mode, using dxvk, without dxvk, dxwrapper, dxwrapper + dxvk without getting errors.
Previously there was an error with RivaTuner that replaced the device reset function hook and some resources were not released and this error occurred, but ThirteenAG had corrected it.

**Comment by RecklessGlue540** [comment]
> Yes, first attempt I got an error with RivaTuner. I spent more than an hour changing settings, switching between windowed and full screen mode, using dxvk, without dxvk, dxwrapper, dxwrapper + dxvk without getting errors. Previously there was an error with RivaTuner that replaced the device reset function hook and some resources were not released and this error occurred, but ThirteenAG had corrected it.

Tested on a fully vanilla CE install now, no such issue with Riva.
Seems to be a FusionFix issue after all...

**Comment by SA64BD** [comment]
> [GTAIV.EFLC.FusionFix_test_01.zip](https://github.com/ThirteenAG/GTAIV.EFLC.FusionFix/files/13824568/GTAIV.EFLC.FusionFix_test_01.zip)
> 
> @RecklessGlue540 Try if this works, I believe I fixed this problem and another possible one, both related to the device reset failure.

Test without RivaTuner. The problem seems completly fixed for me. TYSM @ThirteenAG @RaphaelK12
